types/nightwatch/index.d.ts

```diff
@@ -1784,6 +2220,76 @@ export interface ClientCommands {
      */
     closeWindow(callback?: (this: NightwatchAPI, result: NightwatchCallbackResult<null>) => void): this;
 
+    /**
+     * Sets the current window state to fullscreen.
+     *
+     * @example
+     * module.exports = {
+     *  'demo Test': function(browser) {
+     *     browser.fullscreenWindow(function(result) {
+     *       console.log(result);
+     *     });
+     *   },
+     *
+     *   'ES6 async demo Test': async function(browser) {
+     *     const result = await browser.fullscreenWindow();
+     *     console.log('result value is:', result.value);
+     *   }
+     * }
+     *
+     */
+    fullscreenWindow(callback?: (this: NightwatchAPI, result: NightwatchCallbackResult<null>) => void): this;
+
+    /**
+     * Hides the window in the system tray. If the window happens to be in fullscreen mode,
+     * it is restored the normal state then it will be "iconified" - minimize or hide the window from the visible screen.
+     *
+     * @example
+     * module.exports = {
+     *  'demo Test': function(browser) {
+     *     browser.minimizeWindow(function(result) {
+     *       console.log(result);
+     *     });
+     *   },
+     *
+     *   'ES6 async demo Test': async function(browser) {
+     *     const result = await browser.minimizeWindow();
+     *     console.log('result value is:', result.value);
+     *   }
+     * }
+     */
+    minimizeWindow(callback?: (this: NightwatchAPI, result: NightwatchCallbackResult<null>) => void): this;
+
+    /**
+     * Opens a new top-level browser window, which can be either a tab (default) or a separate new window.
+     *
+     * This command is only available for W3C Webdriver compatible browsers.
+     *
+     * @example
+     * module.exports = {
+     *  'demo Test': function(browser) {
+     *     // open a new window tab (default)
+     *     browser.openNewWindow(function(result) {
+     *       console.log(result);
+     *     });
```

Comment:
This is not supported by the current definition below. You probably need an additional definition like it was done for `clearValue`, so something like this:
```
    openNewWindow(
        callback?: (this: NightwatchAPI, result: NightwatchCallbackResult<null>) => void,
    ): this;
    openNewWindow(
        type?: windowType,
        callback?: (this: NightwatchAPI, result: NightwatchCallbackResult<null>) => void,
    ): this;
```